src/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -618,6 +619,136 @@ public sealed class SocketsHttpHandler_ConnectionUpgrade_Test : HttpClientTestBa
         }
     }
 
+    public sealed class SocketsHttpHandler_Connect_Test : HttpClientTestBase
+    {
+        protected override bool UseSocketsHttpHandler => true;
+
+        [Fact]
+        public async Task ConnectMethod_Success()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    HttpRequestMessage request = new HttpRequestMessage(new HttpMethod("CONNECT"), url);
+                    request.Headers.Add("Host", String.Format("{0}:{1}", request.RequestUri.IdnHost, request.RequestUri.Port));
```

Comment:
We don't need to use request.RequestUri here.  Doing so is weird, because you're asking the server to CONNECT to itself.  I realize it's just a test, but still, it's confusing.

Just make up a host/port.  E.g. "foo.com:345".

Also, you can just use request.Headers.Host instead of using the Add method.